ui/page/send/layout.go

```diff
@@ -276,11 +278,17 @@ func (pg *Page) toSection(gtx layout.Context) layout.Dimensions {
 					layout.Rigid(func(gtx C) D {
 						return layout.Flex{Axis: layout.Horizontal}.Layout(gtx,
 							layout.Rigid(func(gtx C) D {
-								label := pg.Theme.Body2("Exchange rate not fetched")
+								label := pg.Theme.Body2(pg.exchangeError)
 								label.Color = pg.Theme.Color.Danger
+								if strings.Contains(pg.exchangeError, "Retrying") {
+									label.Color = pg.Theme.Color.Primary
+								}
 								return label.Layout(gtx)
 							}),
 							layout.Rigid(func(gtx C) D {
+								if strings.Contains(pg.exchangeError, "Retrying") {
```

Comment:
Sounds like you need a `pg.isFetchingExchangeRate` bool property, as you have in the mainpage. I see that `pg.exchangeError` is set for each failed attempt but the set error message may not show up on the UI without a `RefreshWindow` call. I think you should display a loader or some brief _`fetching exchange rate`_ text while the fetching is ongoing instead of showing an error message for each failed try. At the end of all tries, `if !pg.isFetchingExchangeRate && (the rate isn't ready)`, you can then show the message I recommended above.